electron/js/preload.js

```diff
@@ -123,8 +123,12 @@ ipcRenderer.once('webapp-loaded', function(sender, config) {
       ipcRenderer.send('loaded');
     });
 
-    amplify.subscribe(z.event.WebApp.LIFECYCLE.RESTART, function() {
-      ipcRenderer.send('wrapper-restart');
+    amplify.subscribe(z.event.WebApp.LIFECYCLE.RESTART, function(update_source) {
+      if (update_source === z.announce.UPDATE_SOURCE.DESKTOP) {
+        ipcRenderer.send('wrapper-restart');
+      } else {
+        ipcRenderer.send('wrapper-reload');
```

Comment:
Yes, we can certainly do that moving forward. Do you want to take the lead on this?